### PR TITLE
chore: Use importlib.metadata to get package version

### PR DESCRIPTION
**Pull request by aahung**
### Description of changes

To get rid of the warning:
> serverless-application-model/tests/validator/test_validator_api.py:16: DeprecationWarning: Accessing jsonschema.__version__ is deprecated and will be removed in a future release. Use importlib.metadata directly to query for jsonschema's version.
    jsonschemaMajorVersion = int(jsonschema.__version__.split(".")[0])

### Checklist

- [ ] Adheres to the [development guidelines](https://github.com/aws/serverless-application-model/blob/develop/DEVELOPMENT_GUIDE.md#development-guidelines)
- [ ] Add/update [transform tests](https://github.com/aws/serverless-application-model/blob/develop/DEVELOPMENT_GUIDE.md#unit-testing-with-multiple-python-versions)
    - [ ] Using correct values
    - [ ] Using wrong values
- [ ] Add/update [integration tests](https://github.com/aws/serverless-application-model/blob/develop/INTEGRATION_TESTS.md)

### Examples?

Please reach out in the comments if you want to add an example. Examples will be 
added to `sam init` through [aws/aws-sam-cli-app-templates](https://github.com/aws/aws-sam-cli-app-templates).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by acristin — APPROVED** [review]

**Review by aaythapa — APPROVED** [review]

**Review by ConnorRobertson — APPROVED** [review]

**Review by xazhao — APPROVED** [review]

File: requirements/dev.txt [hunk 1]
```diff
@@ -29,3 +29,7 @@ mypy~=1.1.0
 boto3-stubs[appconfig,serverlessrepo]>=1.19.5,==1.*
 types-PyYAML~=6.0
 types-jsonschema~=3.2
+
+# Check package version (backport importlib.metadata for python 3.7)
+# https://docs.python.org/3/library/importlib.metadata.html
+importlib_metadata; python_version < '3.8'
```
Comment (hoffa) [inline requirements/dev.txt:35]: EOL can't come quick enough

Comment (aahung) [reply requirements/dev.txt:35]: It has EOL. GH will show a red sign if EOL is missing

Comment (hoffa) [reply requirements/dev.txt:35]: Actually, what if we instead fallback to previous behavior? So we don't add dependencies. I highly doubt they'll remove it before EOL.

Comment (hoffa) [reply requirements/dev.txt:35]: > It has EOL. GH will show a red sign if EOL is missing

https://endoflife.date/python

Comment (aahung) [reply requirements/dev.txt:35]: I don't think it is a big deal and using metadata.version is more future proof. The backport is suggested from the official python doc (see code comment) and it is in dev.txt.

Comment (aahung) [reply requirements/dev.txt:35]: > https://endoflife.date/python

LOL What I am thinking. EOF

**Review by hoffa — APPROVED** [review]

**Pull request merged**